crates/ol/stf/src/update.rs

```diff
@@ -0,0 +1,89 @@
+use strata_acct_types::{AccountId, AcctError, BitcoinAmount, MsgPayload};
+use strata_ledger_types::{AccountTypeState, Coin, IAccountState, IL1ViewState, StateAccessor};
+use strata_snark_acct_sys::{handle_snark_msg, handle_snark_transfer};
+
+use crate::{
+    context::BlockExecContext,
+    error::StfResult,
+    system_handlers::{get_system_msg_handler, get_system_transfer_handler},
+};
+
+/// Sends a message with attached value from one account to another.
+///
+/// Creates a [`Coin`] for the message value and adds it to the recipient's balance.
+/// Routes to system handlers for system accounts, or to account-type-specific handlers.
+///
+/// # Safety
+/// Creates a coin with `new_unchecked` - caller must ensure sender has sufficient balance.
+/// The recipient's `add_balance` implementation must call `safely_consume_unchecked` on the coin.
+pub(crate) fn send_message<S: StateAccessor>(
```

Comment:
Is this function also used for deposits? That is, sending a message from "the bridge" to an account to mint coins. If not, then why is this implemented right now?